src/pkg/aws/ecs/tail.go

```diff
@@ -71,65 +72,57 @@ func getLogGroupIdentifier(arnOrId string) string {
 	return strings.TrimSuffix(arnOrId, ":*")
 }
 
-// This is a workaround for the fact that the StartLiveTail API fails if any of the log groups don't exist yet.
-type pendingStream struct {
-	*simpleStream
-	pendingLogGroups []string
-	cw               *cloudwatchlogs.Client
-}
-
-var _ EventStream = (*pendingStream)(nil)
-
-func (ps *pendingStream) Receive(ctx context.Context) ([]LogEvent, error) {
-	if ps.pendingLogGroups == nil {
-		return ps.simpleStream.Receive(ctx) // blocking
-	}
-
-	select {
-	case e := <-ps.Events(): // blocking
-		return convertEvents(e)
-	case <-ctx.Done():
-		return nil, ctx.Err()
-	case <-time.After(100 * time.Millisecond):
-		// Check if any of the log groups have been created
-		s, err := ps.cw.StartLiveTail(ctx, &cloudwatchlogs.StartLiveTailInput{LogGroupIdentifiers: ps.pendingLogGroups})
-		if err != nil {
-			var resourceNotFound *types.ResourceNotFoundException
-			if errors.As(err, &resourceNotFound) {
-				return nil, nil // ignore the error (keep trying)
-			}
-			return nil, err
-		}
-		// Finally able to tail all log groups!
-		ps.pendingLogGroups = nil
-		ps.simpleStream = &simpleStream{s.GetStream()}
-		return nil, nil
-	}
-}
-
 type EventStream interface {
 	Receive(ctx context.Context) ([]LogEvent, error)
 	Close() error
 }
 
 func (a *AwsEcs) TailLogGroups(ctx context.Context, logGroups ...string) (EventStream, error) {
-	for i, lg := range logGroups {
-		logGroups[i] = getLogGroupIdentifier(lg)
+
+	var cs = collectionStream{
+		ch:    make(chan types.StartLiveTailResponseStream),
+		errch: make(chan error),
+		ctx:   ctx,
 	}
-	s, _, err := a.startTail(ctx, &cloudwatchlogs.StartLiveTailInput{LogGroupIdentifiers: logGroups})
-	if err != nil {
+
+	for _, lg := range logGroups {
+		lgID := getLogGroupIdentifier(lg)
+		s, _, err := a.startTail(ctx, &cloudwatchlogs.StartLiveTailInput{LogGroupIdentifiers: []string{lgID}})
+		if err == nil {
+			cs.Add(s)
+		}
+
+		// Start a goroutine to wait for the log group to be created if the error is resource not found
 		var resourceNotFound *types.ResourceNotFoundException
-		if errors.As(err, &resourceNotFound) {
-			// Some log groups don't exist yet, so we'll wait for them to be created
-			s1, cw, err := a.startTail(ctx, &cloudwatchlogs.StartLiveTailInput{LogGroupIdentifiers: []string{logGroups[0]}})
-			if err != nil {
-				return nil, err
-			}
-			return &pendingStream{&simpleStream{s1}, logGroups, cw}, nil
+		if !errors.As(err, &resourceNotFound) {
+			return nil, err
```

Comment:
Some log groups might already have been `Add`-ed

Reply:
Updated the logic to:
1. Try create all live tail streams, if there is any error, fail, record all success streams and resource not found log group ids
2. Create goroutine to wait for all the resource not found log groups
3. addAndStart all the streams to collectionStream to have the log events fan-in